keras_cv/models/resnet_v2.py

```diff
@@ -99,210 +101,312 @@
 """
 
 
-def BasicBlock(
-    filters, kernel_size=3, stride=1, dilation=1, conv_shortcut=False, name=None
-):
+@keras.utils.register_keras_serializable(package="keras_cv.models.resnet_v2")
+class BasicBlock(keras.layers.Layer):
     """A basic residual block (v2).
+
     Args:
         filters: integer, filters of the basic layer.
-        kernel_size: default 3, kernel size of the bottleneck layer.
-        stride: default 1, stride of the first layer.
-        conv_shortcut: default False, use convolution shortcut if True,
+        kernel_size: integer, kernel size of the bottleneck layer. Defaults to 3.
+        stride: integer, stride of the first layer. Defaults to 1.
+        dilation: integer, the dilation rate to use for dilated convolution. Defaults to 1.
+        conv_shortcut: bool, uses convolution shortcut if `True`. Defaults to `False`.
           otherwise identity shortcut.
-        name: string, block label.
+
     Returns:
       Output tensor for the residual block.
     """
-    if name is None:
-        name = f"v2_basic_block_{backend.get_uid('v2_basic_block')}"
-
-    def apply(x):
-        use_preactivation = layers.BatchNormalization(
-            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_use_preactivation_bn"
-        )(x)
-
-        use_preactivation = layers.Activation(
-            "relu", name=name + "_use_preactivation_relu"
-        )(use_preactivation)
 
-        s = stride if dilation == 1 else 1
+    def __init__(
+        self,
+        filters,
+        kernel_size=3,
+        stride=1,
+        dilation=1,
+        conv_shortcut=False,
+        **kwargs,
+    ):
+        if "name" not in kwargs:
+            kwargs["name"] = f"v2_basic_block_{backend.get_uid('v2_basic_block')}"
+        self.filters = filters
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.dilation = dilation
+        self.stride2 = stride if dilation == 1 else 1
+        self.conv_shortcut = conv_shortcut
+        super().__init__(**kwargs)
+
+        self.bn_preactivation = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name=self.name + "_use_preactivation_bn"
```

Comment:
IMO this "use" prefix is kinda useless. Not sure why it was here in the first place, but I think this can just be _preactivation_bn (same for the relu below)

Reply:
(same for BasicBlock)

Reply:
Won't that mess up the existing checkpoints? Might be best to leave this alone for now